### DISCUSSION TITLE
Why aren't the prevent* methods (set with shouldBeStrict()) applied with factories/new objects?

### DISCUSSION THREAD

**Discussion by legecha** [laravel/framework#45984] [Q&A]
I noticed this when I was writing some tests. If I wanted to do something like:

```
$post = Post::factory()->create();

// some stuff

$view = $this->component(RelatedPosts::class, [
    'parent' => $post,
]);

$view->assertSee($post->formatted_title);
```

It's contrived, but illustrates an example. Using test driven development, `formatted_title` doesn't exist yet. I have `Model::shouldBeStrict()` set in my app service provider and expected an exception to be thrown, but the test passes. I realised that [in this commit](https://github.com/laravel/framework/commit/1f86a99f82c55a837c0242eb24127f40f64ad61c) there's a note in the commit message that it will _Only throw on retrieved models_ (checked with `! $this->wasRecentlyCreated`).

I wondered if anyone can tell me why this is the case? Could something untoward happen if this was enabled for models that have only just been created/created with a factory?

Thanks for anyone's thoughts on the matter!

**Comment by LemarinelNet** [comment]
Same for me !
Tests passes and relations are lazy loaded. In production, I got errors stating that model is lazy loaded !
